src/test/javascript/spec/component/lecture/lecture.component.spec.ts

```diff
@@ -0,0 +1,216 @@
+                            }),
+                        );
+                    },
+                    import: (courseId, lectureId) => {
+                        return of(
+                            new HttpResponse({
+                                body: { id: lectureId } as Lecture,
+                                status: 200,
+                            }),
+                        );
+                    },
+                    delete: () => {
+                        return of(new HttpResponse({ status: 200 }));
+                    },
+                }),
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                lectureComponentFixture = TestBed.createComponent(LectureComponent);
+                lectureComponent = lectureComponentFixture.componentInstance;
+                lectureService = TestBed.inject(LectureService);
+                modalService = TestBed.inject(NgbModal);
+            });
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('should initialize', () => {
+        const findAllSpy = jest.spyOn(lectureService, 'findAllByCourseId');
+
+        lectureComponentFixture.detectChanges();
+        expect(lectureComponent).not.toBe(null);
+
+        expect(findAllSpy).toHaveBeenCalledOnce();
+        expect(findAllSpy).toHaveBeenCalledWith(1);
+        expect(lectureComponent.lectures).toBeArrayOfSize(6);
+    });
+
+    it('should delete lecture', () => {
+        const deleteSpy = jest.spyOn(lectureService, 'delete');
+
+        lectureComponentFixture.detectChanges();
+        lectureComponent.deleteLecture(pastLecture.id!);
+
+        expect(deleteSpy).toHaveBeenCalledOnce();
+        expect(deleteSpy).toHaveBeenCalledWith(pastLecture.id!);
+        expect(lectureComponent.lectures).toBeArrayOfSize(5);
+        expect(lectureComponent.lectures).not.toContain(pastLecture);
+        expect(lectureComponent.filteredLectures).toEqual(lectureComponent.lectures);
+    });
+
+    it('should import lecture', async () => {
+        const promise = new Promise((resolve) => {
+            resolve({ id: 123 } as Lecture);
+        });
+        const openSpy = jest.spyOn(modalService, 'open').mockReturnValue({ result: promise } as NgbModalRef);
+        const importSpy = jest.spyOn(lectureService, 'import');
+
+        lectureComponentFixture.detectChanges();
+        lectureComponent.openImportModal();
+
+        expect(openSpy).toHaveBeenCalledOnce();
+        expect(openSpy).toHaveBeenCalledWith(LectureImportComponent, { size: 'lg', backdrop: 'static' });
+
+        await expect(promise)
+            .toResolve()
+            .then(() => {
+                expect(importSpy).toHaveBeenCalledOnce();
+                expect(importSpy).toHaveBeenCalledWith(1, 123);
+                expect(lectureComponent.lectures).toBeArrayOfSize(7);
+            });
+    });
+
+    it('should show all lectures sorted', () => {
```

Comment:
Correct me if I'm wrong, but this test doesn't sort anything, but rather tests the filters.

Reply:
The sorting happens automatically after filtering. After the mock service above returns the lectures in a shuffled order, the filtered lectures should be sorted by id.